examples/toy/Ch4/mlir/ToyCombine.td

```diff
@@ -0,0 +1,26 @@
+
+#ifdef TOY_COMBINE
+#else
+#define TOY_COMBINE
+
+#ifdef OP_BASE
+#else
+include "mlir/IR/OpBase.td"
+include "include/toy/Ops.td"
+#endif // OP_BASE
+
+// Transpose(Transpose(x)) = x
+def TransposeOptPattern : Pat<(TransposeOp(TransposeOp $arg)), (replaceWithValue $arg)>;
+
+// Reshape(Reshape(x)) = x
+def ReshapeReshapeOptPattern : Pat<(ReshapeOp(ReshapeOp $arg)), (ReshapeOp $arg)>;
+
+// Reshape(Constant(x)) = x'
+def ReshapeConstant : NativeCodeCall<"reshapeConstant($_builder, $0)">;
+def FoldConstantReshapeOptPattern : Pat<(ReshapeOp:$res (ConstantOp $arg)), (ReshapeConstant $res)>;
```

Comment:
Here we should be able to pass in $arg and $res, following the comment on the .cpp file:

ReshapeConstant $arg, $res

WDYT?